### PR TITLE
Fix infinite loop time step rejection

### PR DESCRIPTION
**Pull request by wthrowe**
### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: tests/Unit/Time/Actions/Test_ChangeStepSize.cpp [hunk 1]
```diff
@@ -138,32 +143,31 @@ void check(const bool time_runs_forward,
...
+       AdaptiveSteppingDiagnostics{1, 2, 3, 4, 5}, std::move(history), 1.});
 
   ActionTesting::set_phase(make_not_null(&runner), Parallel::Phase::Testing);
   runner.template next_action<component>(0);
   const auto& box = ActionTesting::get_databox<component>(runner, 0);
 
   const size_t index =
       ActionTesting::get_next_action_index<component>(runner, 0);
-  if (reject_step) {
+  if (rejector.has_value()) {
```
Comment (knelli2) [inline tests/Unit/Time/Actions/Test_ChangeStepSize.cpp:171]: You just moved the unique ptr out of the rejector above if it had a value. Does it still contain a value, but just a `nullptr`?

Comment (wthrowe) [reply tests/Unit/Time/Actions/Test_ChangeStepSize.cpp:171]: Yes, that's correct.

**Review by knelli2 — CHANGES_REQUESTED** [review]
@geoffrey4444 could you please try this change out on the case you found that fails? Once you approve, I'll go ahead and merge

**Comment by wthrowe** [comment]
Updated to not test the ASSERT when assertions are not enabled.

**Review by knelli2 — APPROVED** [review]
@geoffrey4444 Has tested this on a BBH and confirms it works

**Pull request merged**